### ISSUE TITLE
关于损失权重的疑问

### ISSUE THREAD

**Issue by zhcm** [ceciliavision/perceptual-reflection-removal#4]
论文中关于损失权重的描述为 L(θ) = 0.1Lfeat(θ) + 0.01Ladv(θ) + Lexcl(θ)；
但是在代码中我注意到实际上使用的损失如下：
```
loss=loss_l1_r+loss_percep*0.2+loss_grad
g_opt=tf.train.AdamOptimizer(learning_rate=0.0002).minimize(loss*100+g_loss, var_list=g_vars)
d_opt=tf.train.AdamOptimizer(learning_rate=0.0001).minimize(d_loss,var_list=d_vars)
```
从实际结果看，g_loss相对loss会比较小，如果在更新g的时候使用loss*100+g_loss，那么g_loss对网络的影响会不会变得几乎没有了呢？

**Comment by ceciliavision** [comment]
L1 loss是在[0,1] normalized后的图片上计算的 所以值会很小 加一个100的权重主要是为了平衡GAN 只是为了做一个数值上的平衡。即使开始有一个loss会相对较大 但毕竟loss在降低 只要scale上不差别很大 还是可以有效的back-prop回去的

**Comment by zhcm** [comment]
我在没修改代码参数、采用您的数据集，训练了大约200个epoch以后的结果显示，数值比较大的是loss_percep部分，loss_percep是在五层vgg feature map上做的L1 loss，或许可能是vgg的feature map数值比较大？
```
    p0 = compute_l1_loss(vgg_real['input'], vgg_fake['input'])
    p1 = compute_l1_loss(vgg_real['conv1_2'], vgg_fake['conv1_2'])/2.6
    p2 = compute_l1_loss(vgg_real['conv2_2'], vgg_fake['conv2_2'])/4.8
    p3 = compute_l1_loss(vgg_real['conv3_2'], vgg_fake['conv3_2'])/3.7
    p4 = compute_l1_loss(vgg_real['conv4_2'], vgg_fake['conv4_2'])/5.6
    p5 = compute_l1_loss(vgg_real['conv5_2'], vgg_fake['conv5_2'])*10/1.5
```
我在训练了200个epoch以后loss_percep达到了70，根据loss=loss_l1_r+loss_percep*0.2+loss_grad，loss的数值大于14，乘100以后已经远远大于g_loss了

**Comment by ceciliavision** [comment]
哦不好意思刚才看漏了一点内容
loss_percep比较大是因为vgg的loss是在[0,255]的图片数值上计算的 达到70是正常的。所以你的问题是现在loss不降么？还是不稳定？
有一个方式是你可以先不用GAN把model train到一个相对稳定的状态 之后再加回去

**Comment by zhcm** [comment]
我尝试了你说的这个方法，但是发现GAN loss仍然没有明显的下降，更新GAN的时候使用的是loss*100+g_loss，似乎感觉数值不太平衡，我现在loss是15，g_loss是10，loss * 100已经是1500了，或许我应该在网络相对稳定之后去掉loss只用g_loss来优化GAN？

**Comment by zhcm** [comment]
啊不好意思，虽然loss_percep很大，但是因为vgg使用[0,255]的图片计算，所以梯度并没有很大的scale，loss和g_loss看起来很大的scale并没有什么影响，现在问题只剩下虽然我尝试了先不用GAN之后再加回去的方法，但是g_loss仍然不降，我是否应该去掉loss只用g_loss来优化呢？

**Comment by ceciliavision** [comment]
我不建议只用gan来train 没有了l1作为constraint会出现一些artifacts 
g_loss只要跟d_loss平衡就好 不过g_loss如果是10有点太大了 那d_loss很小么？ 那图片效果怎样呢

**Comment by zhcm** [comment]
d_loss已经基本为0了，如果整个反射层很平滑的话效果还是可以的；在合成数据集上效果要比真实数据集好一些；如果反射层有一些很明显的边缘的话无论是合成集还是真实集去除效果都很不理想

**Comment by zhanglijian** [comment]
@zhcnn 你好。
 我翻墙下载pre-trained.tar,tz 总是下载失败，能否提供一个资源链接，比如百度网盘！万分感谢
